src/backend/web/graphql/index.js

```diff
@@ -94,7 +122,27 @@ module.exports.resolvers = {
         const last = first + prPage < numOFPosts ? first + prPage : numOFPosts;
         const postIds = await getPosts(first, last);
 
-        return Promise.all(postIds.map(id => Post.byId(id)));
+        const result = await Promise.all(postIds.map(id => Post.byId(id)));
+
+        if (filter) {
+          // check if author name is equal to what we're searching for
+          if (filter.author) {
+            return result.filter(post => post.author === filter.author);
+          }
+          // check if post's date is between two provided dates
+          if (filter.fromDate || filter.toDate) {
+            const fromDate = filter.fromDate ? filter.fromDate : new Date();
+            const toDate = filter.toDate ? filter.toDate : new Date();
+            const datePostIds = await getPostsByDate(fromDate, toDate);
+            return result.filter(post => datePostIds.includes(post.id));
+          }
+          // check if url is equal to what we're searching for
+          if (filter.url) {
+            return result.filter(post => post.url === filter.url);
```

Comment:
We should call the code in `normalize-url` here, so the URLs match more closely.  See https://github.com/Seneca-CDOT/telescope/blob/11700c087602f83838a9a9a67ffe403f8db2cce1/src/backend/data/feed.js#L1